### PR TITLE
task_DEV-3

### PR DESCRIPTION
**Pull request by DmitryZapryvaryn**

### CODE REVIEW

File: Task_03/NumberBaseConverter.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Text;
+
+namespace Task_03
+{
+    /// <summary>
+    /// Provide a method for converts number's base
```
Comment (AlexanderSoroka) [inline Task_03/NumberBaseConverter.cs:8]: Not accurate class description

File: Task_03/NumberBaseConverter.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                return resultNumber.Append(0);
+            }
+
+            uint remainder;
+            while(sourceNumber > 0)
+            {
+                remainder = sourceNumber % newBase;
+                if(remainder > 9)
+                {
+                    resultNumber.Insert(0, (char)(remainder + 55));
```
Comment (AlexanderSoroka) [inline Task_03/NumberBaseConverter.cs]: why +55?

File: Task_03/NumberBaseConverter.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                }
+                sourceNumber /= newBase;
+            }
+
+            return resultNumber;
+        }
+
+        private bool IsValidBase(string stringBase, out uint result)
+        {
+            return UInt32.TryParse(stringBase, out result) && (result <= 20 && result >= 2);
```
Comment (AlexanderSoroka) [inline Task_03/NumberBaseConverter.cs]: single responsibility violation, wrong error-handling approach

File: Task_03/NumberBaseConverter.cs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            if (args.Length != 2)
+            {
+                Console.WriteLine("Run program with two arguments: number, base.");
+            }
+            else
+            {
+                NumberBaseConverter radixChanger = new NumberBaseConverter();
+                uint inputNumber;
+                uint newBase;
+                bool isDecimalNumber = UInt32.TryParse(args[0], out inputNumber);
```
Comment (AlexanderSoroka) [inline Task_03/NumberBaseConverter.cs]: wrong error-handling approach

File: Task_03/NumberBaseConverter.cs [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        private const byte AsciiStartChar = 55;
+
+        /// <summary>
+        /// The method converts number from decimal base to new base. 
+        /// </summary>
+        /// <param name="sourceNumber">A number which need to convert to other base.</param>
+        /// <param name="newBase">New numeric system.</param>
+        /// <returns>Number in new numeric system as StringBuilder.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">If new numeric system have illegal value( new base less then 2 or more than 20).</exception>
+        /// <exception cref="ArgumentException">Source number must be positive</exception>
```
Comment (IgorKheidorov) [inline Task_03/NumberBaseConverter.cs]: Not changed number, number is the same, only the presentation changed

File: Task_03/NumberBaseConverter.cs [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+            StringBuilder resultNumber = new StringBuilder();
+            if (sourceNumber.IsZero)
+            {
+                return resultNumber.Append(0);
+            }
+
+            int remainder;
+            while(sourceNumber > 0)
+            {
+                remainder = (int)(sourceNumber % newBase);
```
Comment (IgorKheidorov) [inline Task_03/NumberBaseConverter.cs]: can be simplified in one line

**Review by IgorKheidorov — CHANGES_REQUESTED** [review]

File: Task_03/NumberBaseConverter.cs [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+using System.Numerics;
+
+namespace Task_03
+{
+    /// <summary>
+    /// Provide a method for converts number's base
+    /// </summary>
+    class NumberBaseConverter
+    {
+        //If base of new number > 10 we need to use char A,B,C... in number.
```
Comment (AlexanderSoroka) [inline Task_03/NumberBaseConverter.cs:12]: useless comments, code must be self-descriptive

File: Task_03/NumberBaseConverter.cs [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+        //65 in ASCII is 'A'.
+        //Char code = 65 + (remainder - 10) => char code = 55 + remainder.  
+        private const byte AsciiStartChar = 55;
```
Comment (AlexanderSoroka) [inline Task_03/NumberBaseConverter.cs:15]: not obvious constant name

File: Task_03/NumberBaseConverter.cs [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+        /// <exception cref="ArgumentOutOfRangeException">If new numeric system have illegal value( new base less then 2 or more than 20).</exception>
+        /// <exception cref="ArgumentException">Source number must be positive</exception>
+        public StringBuilder GetNewNumberPresentation(BigInteger sourceNumber, int newBase)
+        {
+            if (newBase < 2 || newBase > 20)
+            {
+                throw new ArgumentOutOfRangeException("newBase");
+            }
+
+            if(sourceNumber < 0)
```
Comment (AlexanderSoroka) [inline Task_03/NumberBaseConverter.cs]: codestyle violation

File: Task_03/NumberBaseConverter.cs [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+                throw new ArgumentException("Source number must be positive", "sourceNumber");
+            }
+
+            StringBuilder resultNumber = new StringBuilder();
+            if (sourceNumber.IsZero)
+            {
+                return resultNumber.Append(0);
+            }
+
+            while(sourceNumber > 0)
```
Comment (AlexanderSoroka) [inline Task_03/NumberBaseConverter.cs]: too long method

File: Task_03/NumberBaseConverter.cs [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+        //If base of new number > 10 we need to use char A,B,C... in number.
+        //65 in ASCII is 'A'.
+        //Char code = 65 + (remainder - 10) => char code = 55 + remainder.  
+        private const byte AsciiStartChar = 55;
+
+        /// <summary>
+        /// The method converts number from decimal base to new base. 
+        /// </summary>
+        /// <param name="sourceNumber">A number which need to convert to other base.</param>
+        /// <param name="newBase">New numeric system.</param>
```
Comment (AlexanderSoroka) [inline Task_03/NumberBaseConverter.cs:21]: Inaccurate param description

**Review by AlexanderSoroka — APPROVED** [review]
Approved due to deadline

**Pull request merged**